core/src/main/scala/org/apache/spark/sql/delta/GeneratedColumn.scala

```diff
@@ -528,3 +515,19 @@ object GeneratedColumn extends DeltaLogging with AnalysisHelper {
   private val DATE_FORMAT_YEAR_MONTH = "yyyy-MM"
   private val DATE_FORMAT_YEAR_MONTH_DAY_HOUR = "yyyy-MM-dd-HH"
 }
+
+object AttributeOrNested {
+  def unapply(e: Expression): Option[(String, DataType)] = e match {
+    case AttributeReference(name, dataType, _, _) =>
+      Some((escapeName(name), dataType))
+    case g @ GetStructField(child, ordinal, _) => child match {
+      case AttributeOrNested(name, _) =>
+        val fieldName = child.dataType.asInstanceOf[StructType].fieldNames(ordinal)
+        Some((Seq(name, escapeName(fieldName)).mkString("."), g.dataType))
+      case _ => None
+    }
+    case _ => None
+  }
+
+  def escapeName(colName: String): String = if (colName.contains(".")) s"`$colName`" else colName
```

Comment:
Took the logic from `UnresolvedAttribute`

Reply:
and if it's necessary add a comment explaining why

Reply:
Since we're converting the all the fields to a string, periods are ambiguous if they're allowed in field names. `outer.inner.value` could be a top level field called `outer.inner`, for example, so in that case it makes it 
```
`outer.inner`.value
```
just trying to be extra safe, didn't actually test if periods are allowed in delta field names

Reply:
added a comment

Reply:
All these names are pulled from the schema though. I'll see if I can put together an example of when this could be an issue. I was mostly trying to just be extra safe, I don't personally use columns with periods in them hah

Reply:
Makes sense, it would be nice to have a negative test case

Reply:
I added a test case showing what that is there for (obviously very edge case-y)

Reply:
Can you also make this support backtick itself by calling `org.apache.spark.sql.catalyst.util.quoteIfNeeded`?

Reply:
I can just replace my escaping with that right?